### PR TITLE
Add search bar 🔍 | Split symbols out

### PR DESCRIPTION
**Pull request by CatalanCabbage**
# Split symbols out of the HTML
```js
symbols = {'©', '®', '™'} //has been split into
symbols = {"glyph": "²", "name": "Squared", "searchTerms": ["squared", "power"]}, //etc
```

- No need to worry about symbol contributions breaking HTML
- We can add metadata like `name`, `searchTerms` etc which brings me to....

# Add a rudimentary search bar
A simple search bar has been added which performs a naive search across the symbols.  
It takes data from the array `searchTerms`, concats them and checks for a partial match.  
Should be good enough for a long time. Unless we want _AI and ML and 100,000 symbols_. :)

_Considering `searchTerms` is a separate value, it's full customizable - ✔ will show up on searches for `check`, `tick`, `correct` etc - go wild!_

> _Comments on style_
> Kept the default since it's in line with the general style of the page. Just centered it (WHICH TOOK ME 3 HOURS)
> btw the search results view is endearing - becomes bigger in size with fewer results. Pretty cool, so I let it be.

-----
# Now here's what you came for
_the search!_
image

_ooh only 2 hits so they're chonky boisss_
image

### CODE REVIEW

**Comment by qwell** [comment]
#17 no longer made sense given this PR, but I would like to push a bit for the way it did categories, even if only via comments for now.  It would be nice to scroll through things in a semi-coherent list and it could help with searching as well.

File: symbols.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
+const symbols = [
+    {
+        "glyph": "©",
+        "name": "Copyright",
+        "searchTerms": ["copyright"]
+    },
+    {
+        "glyph": "®",
+        "name": "Registered Trademark",
+        "searchTerms": ["registered trademark"]
```
Comment (qwell) [inline symbols.js:10]: should probably be `["registered", "trademark"]`, and maybe add "r"

File: symbols.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    },
+    {
+        "glyph": "™",
+        "name": "Trademark",
+        "searchTerms": ["trademark"]
```
Comment (qwell) [inline symbols.js:15]: add "tm"

File: symbols.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    },
+    {
+        "glyph": "…",
+        "name": "Ellipsis",
+        "searchTerms": ["ellipsis"]
+    },
+    {
+        "glyph": "½",
+        "name": "Half",
+        "searchTerms": ["half"]
```
Comment (qwell) [inline symbols.js:40]: add "1/2"

File: symbols.js [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    },
+    {
+        "glyph": "¼",
+        "name": "Quarter",
+        "searchTerms": ["quarter"]
```
Comment (qwell) [inline symbols.js:45]: add "1/4"

File: index.html [hunk 5]
```diff
@@ -96,17 +104,18 @@
...
-
-    document.addEventListener("DOMContentLoaded", () => {
+    function renderSymbols(searchTerm = "") {
       const parent = document.querySelector(".symbols");
-      for (const symbol of symbols) {
+      searchTerm = searchTerm.toLowerCase();
+      parent.innerHTML = "";
+      for (const symbolInfo of symbols) {
+        symbolSearchTerms = symbolInfo.searchTerms.join(" ");
+        if (searchTerm != "" && !symbolSearchTerms.toLowerCase().includes(searchTerm)) {
```
Comment (qwell) [inline index.html:113]: `!==`

File: index.html [hunk 6]
```diff
@@ -115,16 +124,34 @@
           elem.classList = "symbol-clicked";
```
Comment (qwell) [inline index.html:124]: `elem.classList.add("symbol-clicked")`

File: index.html [hunk 7]
```diff
@@ -115,16 +124,34 @@
...
 
           setTimeout(() => {
-            elem.textContent = symbol;
+            elem.textContent = symbolInfo.glyph;
+            elem.title = symbolInfo.name;
             elem.classList = "symbol";
```
Comment (qwell) [inline index.html:129]: `elem.classList.remove("symbol-clicked")`, maybe?

File: symbols.js [hunk 8]
```diff
@@ -0,0 +1,142 @@
+const symbols = [
+    {
+        "glyph": "©",
+        "name": "Copyright",
+        "searchTerms": ["copyright"]
```
Comment (qwell) [inline symbols.js:5]: "c"

File: symbols.js [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+    },
+    {
+        "glyph": "™",
+        "name": "Trademark",
+        "searchTerms": ["trademark"]
+    },
+    {
+        "glyph": "“",
+        "name": "Left Double Quotation Mark",
+        "searchTerms": ["quotation", "quote", "double"]
```
Comment (qwell) [inline symbols.js:20]: Maybe add `"` here and below?

**Review by qwell — CHANGES_REQUESTED** [review]

**Comment by qwell** [comment]
Created PR downstream to address my comments here.

https://github.com/CatalanCabbage/symbol.wtf/pull/1

**Comment by qwell** [comment]
Add several more downstream PRs to add some extra functionality.

1) Invisible characters (https://github.com/CatalanCabbage/symbol.wtf/pull/2)
2) Autofocus (https://github.com/CatalanCabbage/symbol.wtf/pull/3).

**Comment by samwho** [comment]
I'm going to merge this into a staging branch and apply the review comments from @qwell. Thank you both for your contributions :)

**Comment by qwell** [comment]
If you give me a minute, I'll get my rollup branch pointed to yours and open a separate PR.

**Comment by samwho** [comment]
Ahh sorry, I already did the merge. If you rebase off `main` and open a new PR I'll take a look 🙏

**Comment by qwell** [comment]
> Ahh sorry, I already did the merge. If you rebase off `main` and open a new PR I'll take a look 🙏

#39 :)

**Pull request merged**